### PR TITLE
Enhanced Backend Functionality for Service Authorization

### PR DESCRIPTION
**Pull request by franklbh**
Added a new backend function authClient that:

- Accepts an oidcConfig object with clientId, clientSecret, scope, and optional authority.

- Instantiates a ServiceAuthorizationClient using the provided configuration.

- Fetches an access token using the client credentials flow and returns it.

Registered the new backend function with registerBackendCallback under the name "getServiceAuthToken"

### CODE REVIEW

File: packages/oidc-signin-tool/src/certa/certaBackend.ts [hunk 1]
```diff
@@ -56,12 +57,43 @@ async function signin(user: TestUserCredentials, oidcConfig?: TestBrowserAuthori
   if (undefined === token)
     throw new Error("Failed to get access token");
 
+
   return token;
 }
 
+async function signinWithServiceAuthClient(oidcConfig: ServiceAuthorizationClientConfiguration): Promise<string> {
+  // const clientId = process.env.IMJS_AGENT_TEST_CLIENT_ID;
```
Comment (hl662) [inline packages/oidc-signin-tool/src/certa/certaBackend.ts:65]: remove the commented out code

File: package.json [hunk 2]
```diff
@@ -33,9 +33,9 @@
     "url": "https://github.com/iTwin/auth-clients"
   },
   "devDependencies": {
-    "beachball": "^2.43.1",
-    "lage": "^2.7.18",
-    "path-browserify": "^1.0.0",
+    "beachball": "^2.49.1",
+    "lage": "^2.11.13",
+    "path-browserify": "^1.0.1",
```
Comment (aruniverse) [inline package.json:38]: do we still need this dep? if so, what for? same thing with process

File: packages/oidc-signin-tool/src/certa/certaBackend.ts [hunk 3]
```diff
@@ -7,9 +7,10 @@ import * as fs from "fs";
 import * as path from "path";
 import type { AccessToken } from "@itwin/core-bentley";
 import { registerBackendCallback } from "@itwin/certa/lib/utils/CallbackUtils";
+import { ServiceAuthorizationClient, ServiceAuthorizationClientConfiguration } from "@itwin/service-authorization";
```
Comment (aruniverse) [inline packages/oidc-signin-tool/src/certa/certaBackend.ts:10]: where is this dep defined?

File: packages/oidc-signin-tool/src/certa/certaBackend.ts [hunk 4]
```diff
@@ -56,12 +57,43 @@ async function signin(user: TestUserCredentials, oidcConfig?: TestBrowserAuthori
...
 registerBackendCallback(
   getTokenCallbackName,
   async (user: any, oidcConfig?: any): Promise<string> => {
     return signin(user, oidcConfig);
   },
 );
+
+
+registerBackendCallback(getServiceAuthTokenCallbackName, async (oidcConfig: ServiceAuthorizationClientConfiguration): Promise<string> => {
+  return await signinWithServiceAuthClient(oidcConfig);
```
Comment (hl662) [inline packages/oidc-signin-tool/src/certa/certaBackend.ts:98]: do you need the await here?

File: packages/oidc-signin-tool/src/certa/certaBackend.ts (hunk 3 above)
Comment (hl662) [reply packages/oidc-signin-tool/src/certa/certaBackend.ts:10]: it's not a dev dependency, if that's what you are thinking of...?

Comment (aruniverse) [reply packages/oidc-signin-tool/src/certa/certaBackend.ts:10]: nvm, i see it as a direct dep thats fine.
was surprised to see a new import w/o a pkg.json update

File: packages/oidc-signin-tool/src/certa/certaCommon.ts [hunk 5]
```diff
@@ -4,9 +4,11 @@
 *--------------------------------------------------------------------------------------------*/
 import { executeBackendCallback } from "@itwin/certa/lib/utils/CallbackUtils";
 import type { TestBrowserAuthorizationClientConfiguration, TestUserCredentials } from "../TestUsers";
+import { ServiceAuthorizationClientConfiguration } from "@itwin/service-authorization";
```
Comment (aruniverse) [inline packages/oidc-signin-tool/src/certa/certaCommon.ts:7]: should prob be import type

File: packages/oidc-signin-tool/src/certa/certaBackend.ts (hunk 1 above)
Comment (franklbh) [reply packages/oidc-signin-tool/src/certa/certaBackend.ts]: > remove the commented out code

just did

File: packages/oidc-signin-tool/src/certa/certaCommon.ts (hunk 5 above)
Comment (franklbh) [reply packages/oidc-signin-tool/src/certa/certaCommon.ts]: > should prob be import type

just did it

File: packages/oidc-signin-tool/src/certa/certaBackend.ts (hunk 4 above)
Comment (franklbh) [reply packages/oidc-signin-tool/src/certa/certaBackend.ts]: no its redundant, just deleted it

**Review by hl662 — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

**Review by ben-polinsky — APPROVED** [review]
LGTM, not sure what the pipeline failure is about

**Pull request merged**